pcdsdevices/pim.py

```diff
@@ -30,39 +28,197 @@ class PIMMotor(InOutRecordPositioner):
+        """Moves the diode into the beam."""
+        return self.state.insert(moved_cb=moved_cb, timeout=timeout,
+                                 wait=wait)
+
+    def remove(self, moved_cb=None, timeout=None, wait=False):
+        """Moves the diode out of the beam."""
+        return self.state.remove(moved_cb=moved_cb, timeout=timeout,
+                                 wait=wait)
+
+    def __init__(self, prefix, *, name, prefix_det=None, prefix_zoom=None,
+                 **kwargs):
+        self.infer_prefix(prefix)
+
+        # Infer the detector PV from the base prefix
+        if prefix_det:
             self._prefix_det = prefix_det
+        else:
+            self._prefix_det = self.prefix_start+'CVV:01'
+
+        # Infer the zoom motor PV from the base prefix
+        if prefix_zoom:
+            self._prefix_zoom = prefix_zoom
+        else:
+            self._prefix_zoom = self.prefix_start+'CLZ:01'
+
+        super().__init__(prefix, name=name, **kwargs)
+        self.y_motor = self.state.motor
+
+
+class PIM_withFocus(PIM):
+    """
+    Profile intensity monitor with y-motion motor, zoom motor, focus motor, and
+    a detector.
+
+    Parameters
+    ----------
+    prefix : str
+        The EPICS base of the PIM
+
+    name : str
+        A name to refer to the device
+
+    prefix_det : str, optional
+        The EPICS base PV of the detector. If None, it will be attemptted to be
+        inferred from `prefix`
+
+    prefix_zoom : str, optional
+        The EPICS base PV of the zoom motor. If None, it will be attempted to
+        be inferred from `prefix`
+
+    prefix_focus : str, optional
+        The EPICS base PV of the focus motor. If None, it will be attempted to
+        be inferred from `prefix`
+    """
+    focus_motor = FCpt(IMS, '{self._prefix_focus}')
+
+    def __init__(self, prefix, *, name, prefix_focus=None, **kwargs):
+        self.infer_prefix(prefix)
+
+        # Infer the focus motor PV from the base prefix
+        if prefix_focus:
+            self._prefix_focus = prefix_focus
+        else:
+            self._prefix_focus = self.prefix_start+'CLF:01'
+
         super().__init__(prefix, name=name, **kwargs)
+
+
+class PIM_withLED(PIM):
+    """
+    Profile intensity monitor with y-motion motor, zoom motor, LED, and a
+    detector.
+
+    Parameters
+    ----------
+    prefix : str
+        The EPICS base of the PIM
+
+    name : str
+        A name to refer to the device
+
+    prefix_det : str, optional
+        The EPICS base PV of the detector. If None, it will be attemptted to be
+        inferred from `prefix`
+
+    prefix_zoom : str, optional
+        The EPICS base PV of the zoom motor. If None, it will be attempted to
+        be inferred from `prefix`
+
+    prefix_led : str, optional
+        The EPICS base PV of the LED. If None, it will be attempted to be
+        inferred from `prefix`
+    """
+    led = FCpt(EpicsSignal, '{self._prefix_led}')
+
```

Comment:
```suggestion
tab_whitelist = ['led']

```

Reply:
Agreed on the suggestion, but be careful accepting it as the indentation is wrong.

Reply:
my b

Reply:
also, I just remembered: you should be able to set `tab_component_names` to `True` to automatically get all of the components tabbable

Reply:
Used tab_component_names in new commit.